src/miral/cursor_scale.cpp

```diff
@@ -24,26 +24,34 @@
 
 struct miral::CursorScale::Self
 {
-    Self() {}
-
-    Self(float default_scale): _scale{default_scale} {}
+    Self(float default_scale) :
+        default_scale{default_scale}
+    {
+    }
 
     void scale(float new_scale)
     {
-        _scale = new_scale;
+        auto const locked_state = state.lock();
+        locked_state->scale.emplace(new_scale);
 
-        if(accessibility_manager.expired())
+        if(locked_state->accessibility_manager.expired())
             return;
 
-        accessibility_manager.lock()->cursor_scale_changed(*_scale);
+        locked_state->accessibility_manager.lock()->cursor_scale_changed(*locked_state->scale);
     }
 
-    std::optional<float> _scale;
-    std::weak_ptr<mir::shell::AccessibilityManager> accessibility_manager;
+    float const default_scale;
+
+    struct State {
+        std::optional<float> scale;
+        std::weak_ptr<mir::shell::AccessibilityManager> accessibility_manager;
+    };
+
+    mir::Synchronised<State> state;
```

Comment:
As discussed, there's no need to guard `accessibility_manager`, once multiple threads have started it is effectively "const".

There is no need for a separate `default_scale` in addition to `scale`.

There is a need to guard `scale`, but `Synchronised<State>` is likely overkill.

```suggestion

    std::atomic<float> scale;
    
    // accessibility_manager is only updated during the single-threaded initialization phase of startup
    std::weak_ptr<mir::shell::AccessibilityManager> accessibility_manager;
```